transient/config.py

```diff
@@ -0,0 +1,261 @@
+"""Supports the creation and validation of Transient-run configurations
+"""
+
+import toml
+import sys
+
+from marshmallow import Schema, fields, post_load, ValidationError
+
+from typing import Any, Dict, MutableMapping
+
+
+default_config_file = """# A Transient config file
+
+#
+# Transient arguments are set in a similar manner compared to the command line.
+# A few differences exist:
+#
+#   1. All values are set using an equal sign:
+#
+#          shutdown-timeout = 20
+#
+#
+#   2. Strings are encapsulated with double quotes
+#
+#          name = "vm_name"
+#
+#
+#   3. Flags are set to Booleans:
+#
+#          prepare-only = true
+#          ssh-console = false
+#
+#
+#   4. Lists are encapsulated by square brackets:
+#
+#          shared-folder = [
+#              "/path/on/host:/path/on/guest",
+#              "/path/on/host2:/path/on/guest2"
+#          ]
+#
+
+[transient]
+
+copy-in-before   = []
+copy-out-after   = []
+copy-timeout     = 0
+image            = ["centos/7:2004.01"]
+image_backend    = ""
+image_frontend   = ""
+name             = "vm_name"
+prepare-only     = false
+qmp-timeout      = 10
+shared-folder    = []
+shutdown-timeout = 20
+ssh-bin-name     = ""
+ssh-command      = ""
+ssh-console      = false
+ssh-port         = 22
+ssh-user         = "vagrant"
+ssh-with-serial  = false
+
+
+#
+# QEMU arguments are set in a similar manner compared to the command line.
+#
+# The only difference is that the QEMU argument string is split by whitespace
+# and encapsulated in a list.
+#
+
+[qemu]
+
+qemu-args = [
+    "-nographic",
+    "-enable-kvm",
+    "-m", "1G"
+]
+"""
+
+
+class Config(Dict[Any, Any]):
```

Comment:
Isn't this exactly what argparse.Namespace is? Maybe just make an alias for that instead of this? We aren't really getting anything from the `Dict[Any, Any]` type info anyway.

Reply:
The behavior is the same as far as I can tell, which I intended. I saw that you used `argparse.Namespace` directly in `cli.py` but I couldn't find documentation on how to directly use the class. The official docs indicate that it's only used by `parse_args()`: 
 https://docs.python.org/3.8/library/argparse.html#namespace 

Additionally, I have to combine arguments from the CLI and the configuration file - I'm not sure how I would accomplish that with using `argparse.Namespace`. 

Ultimately, `argparse.Namespace` is opaque to me and it seems, to my knowledge, only used by the `argparse` library. If I'm missing something, feel free to point me in the right direction and I can remove this class.

Reply:
I've pushed my latest changes but this is still unresolved. Let me know how you want to move forward with this.

Reply:
I'm going to mark this as resolved. But if you want me to replace this with `argparse.Namespace`, please let me know.